File: python/sample/dep_parser_usage.py
from koalanlp.Util import initialize
from koalanlp.proc import Parser
from koalanlp import API

initialize(KKMA='LATEST')  #: HNN=2.0.4, ETRI=2.0.4

parser = Parser(API.KKMA)

while True:
    text = input("분석할 문장을 입력하세요>> ").strip()

    if len(text) == 0:
        break

    sentences = parser(text)

    for sent in sentences:
        print("===== Sentence =====")
        print(sent.singleLineString())
        print("# Dependency Parse result")

        dependencies = sent.getDependencies()
        if len(dependencies) > 0:
            for edge in dependencies:
                print("[%s]는 [%s]의 %s-%s" % (edge.getDependent().getSurface(),
                                             edge.getGovernor().getSurface() if edge.getGovernor() is not None else "ROOT",
                                             str(edge.getType()),
                                             str(edge.getDepType())))
        else:
            print("(Unexpected) NULL!")

File: nodejs/sample/depParserUsage.js
const {initialize} = require("koalanlp/Util");
const {Parser} = require("koalanlp/proc");
const {KKMA} = require("koalanlp/API");
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let parser;

async function processText(text){
    if (text == null || text.trim().length === 0) {
        rl.close();
        return;
    }

    let sentences = await parser(text);
    sentences.forEach((sent, i) => {
        console.log(`===== Sentence #${i} =====`);
        console.log(sent.singleLineString());

        console.log("# Dependency Parse result");
        sent.getDependencies().forEach((edge) => {
            console.log(`[${edge.dependent.surface}]는 [${edge.governor ? edge.governor.surface : "ROOT"}]의 ${edge.type}-${edge.depType}`);
        });
    });

    readInput();
}

function readInput(){
    rl.question("분석할 문장을 입력하세요>> ", processText);
}

/***********************
 * Main Execution Part *
 ***********************/
initialize({packages: {KKMA: 'LATEST'}})  //ETRI, HNN
    .then(() => {
        parser = new Parser(KKMA);
        readInput();
    }).catch((err) => console.error('Error occurred!', err));